### PR TITLE
Playwright test: User should be able to create account

### LINKED ISSUE #37: Implement test for user account creation
- [ ] User should be able to create an account

### PR DESCRIPTION
**Pull request by hariclerry**
### Tickets
Closes #37 

### What does this PR do?
This PR implements user account creation test in playwright.

### What are the changes made?
- Write user account creation test.

### Screen shots
None

### CODE REVIEW

**Comment by hariclerry** [comment]
Hey @und3fined-v01d, I will delete/remove the selenium create user account test after your review of this PR. Thanks

File: tests/playwright/helpers/Util.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+function getTestString( prefix = '' ) {
```
Comment (vidhi-mody) [inline tests/playwright/helpers/Util.js]: Please use ES6 functions wherever possible.

File: tests/playwright/jest.e2e.config.js [hunk 2]
```diff
@@ -1,4 +1,12 @@
+const path = require( 'path' );
+
 module.exports = {
+	globals: {
+		baseUrl:
```
Comment (vidhi-mody) [inline tests/playwright/jest.e2e.config.js:5]: Use all caps with snake-case for constants.

File: tests/playwright/specs/user.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+const { CreateAccountPage } = require( '../pageobjects/index.js' );
```
Comment (vidhi-mody) [inline tests/playwright/specs/user.js]: index.js is not required here.

File: tests/playwright/specs/user.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+const { getTestString } =
+ require( '../helpers/Util.js' );
```
Comment (vidhi-mody) [inline tests/playwright/specs/user.js]: remove the `.js`

**Review by vidhi-mody — CHANGES_REQUESTED** [review]

File: tests/playwright/specs/user.js (hunk 4 above)
Comment (hariclerry) [reply tests/playwright/specs/user.js]: @vidhi-mody, eslint expects the import to be /Util.js. There is an eslint error when I remove the .js

File: tests/playwright/specs/user.js (hunk 3 above)
Comment (hariclerry) [reply tests/playwright/specs/user.js]: @vidhi-mody, eslint expects the import to be /index.js. There is an eslint error when I remove the index.js part.

File: tests/playwright/specs/user.js (hunk 4 above)
Comment (hariclerry) [reply tests/playwright/specs/user.js]: It has been fixed, added eslint lint rule.

File: tests/playwright/specs/user.js (hunk 3 above)
Comment (hariclerry) [reply tests/playwright/specs/user.js]: It has been fixed, added eslint lint rule.

Comment (hariclerry) [reply tests/playwright/specs/user.js]: I have added eslint ignore rule. It should be fine now

File: tests/playwright/specs/user.js (hunk 4 above)
Comment (hariclerry) [reply tests/playwright/specs/user.js]: I have added eslint ignore rule. It should be fine now

File: tests/playwright/pageobjects/page.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+	 * Navigate the browser to a given page.
+	 *
+	 * @param {string} title Page title
+	 * @param {Object} [query] Query parameter
+	 * @param {string} [fragment] Fragment parameter
+	 * @return {void} This method runs a browser command.
+	 */
+	async openTitle( title, query = {}, fragment = '' ) {
+		query.title = title;
+		await page.goto(
```
Comment (sovoid) [inline tests/playwright/pageobjects/page.js:14]: why is it that in some places we use `global.page` while at others we use just `page`

File: tests/playwright/pageobjects/createaccount.page.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+		this.username = '#wpName2';
+		this.password = '#wpPassword2';
+		this.confirmPassword = '#wpRetype';
+		this.create = '#wpCreateaccount';
+		this.heading = '.firstHeading';
+	}
+
+	async createAccount( username, password ) {
+		await this.openTitle( 'Special:CreateAccount' );
+		await global.page.type( this.username, username );
```
Comment (sovoid) [inline tests/playwright/pageobjects/createaccount.page.js:16]: Do you really need a `global` here?

**Review by sovoid — CHANGES_REQUESTED** [review]

File: tests/playwright/specs/user.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+	beforeEach( async () => {
+		username = Util.getTestString( 'User-' );
+		password = Util.getTestString();
+	} );
+
+	it( 'should be able to create account', async () => {
+		await CreateAccountPage.createAccount( username, password );
+		await page.screenshot( {
+			path: `${global.logPath}/userAccountCreation.png`
```
Comment (sovoid) [inline tests/playwright/specs/user.js:16]: This is a lot of hardcoding. Add this to the beforeEach and afterEach for every file. Checkout how we name files for Selenium

**Review by sovoid — CHANGES_REQUESTED** [review]

**Pull request closed without merging**